src/NuGet.Core/NuGet.Packaging/RuntimeModel/IObjectWriter.cs

```diff
@@ -11,6 +12,17 @@ namespace NuGet.RuntimeModel
     /// </summary>
     public interface IObjectWriter
     {
+        /// <summary>
+        /// Writes the start of the root object or an object in an array.
+        ///
+        /// This new object becomes the scope for all other method calls until either WriteObjectStart
+        /// is called again to start a new nested object or WriteObjectEnd is called.
+        ///
+        /// Every call to WriteObjectStart must be balanced by a corresponding call to WriteObjectEnd.
+        /// </summary>
+        /// <exception cref="ObjectDisposedException">Thrown if this object is disposed.</exception>
+        void WriteObjectStart();
```

Comment:
Last time, someone complained about this breaking change: https://github.com/NuGet/Home/issues/8080.

Since then they have fixed their dependency https://github.com/dotnet/arcade/pull/2664/files, but we need to be aware that people do take dependencies. 

We can say, the client tooling comes first though :)

Reply:
We just observed this breaking change in https://github.com/dotnet/runtime/pull/48462 after upgrading the SDK to 6.0. You could argue that our custom packaging code should be replaced by NuGet's pack task, but we aren't there yet.

https://github.com/dotnet/arcade/blob/1ca4a4637bf32494c8ba579c32e4512309cae751/src/Microsoft.DotNet.Build.Tasks.Packaging/src/NuGetUtility.cs#L94

cc @Anipik @ericstj

Reply:
What is interesting about this is, even though the change was merged in Feb 2020, it didn't make it into 5.0.100. Was that intentional? The fix for us in dotnet/arcade is just to reference newer nuget packages to update what we bind against.

Reply:
This change is in 5.0.100, https://github.com/NuGet/NuGet.Client/blob/release-5.8.x/src/NuGet.Core/NuGet.Packaging/RuntimeModel/IObjectWriter.cs

Or at least should be according to the branch and tags.

Can you help me understand the problem you are seeing and why you think it's not in 5.0.100.

Reply:
I think @nkolev92 was commenting that this is breaking because they added a member to an interface.  We aren't broken by that addition.  I think the break we're currently seeing is something different.